### ISSUE TITLE
Error spam when moving the animation timeline cursor when a track is in capture mode

### ISSUE THREAD

**Issue by YeldhamDev** [godotengine/godot#32529]
**Godot version:**
d86c9ef

**Issue description:**
While moving the timeline cursor (the blue line) in the animation editor around the time of 0.0 and the first key, and one of the tracks is in capture mode, it will spam this error:
```
scene/animation/animation_player.cpp:887 - Failed setting key at time 0.1 in Animation '' at Node
'/root/EditorNode/@@580/@@581/@@589/@@591/@@595/@@599/@@600/@@601/@@617/@@618/@@627/@@628/@@6386/@@6246/@@6247/@@6248/@@6249/@@6250/AnimationPlayer',
Track 'TrackName:property_name'. Check if property exists or the type of key is right for the property
```

~~It only started happening recently.~~ I just tested with 3.1.1 and it happens there as well. Guess I never encountered that bug there before somehow.

I tried to revert @KoBeWi's recent changes to that editor to see if it was the one affecting it, but no dice.

**Comment by Relintai** [comment]
For me it does this, when the cursor is between 0 and the first keyframe. 
Or if the first keyframe is at 0, then it does it between that and the second keyframe.

**Comment by Gramps** [comment]
I also get this error and it seems to have started randomly.  It complains about a label that I move but the label still moves perfectly fine.  The error seems to make no sense as the property and key are both correct.

**Comment by akien-mga** [comment]
Is this still reproducible in the current master branch?

**Comment by KoBeWi** [comment]
Yes, tested in 90a224c6e

**Comment by alvaroHernandez** [comment]
Hi, I was taking a look to this. I found the main issue in: `AnimationPlayer::_animation_process_animation:445`

Where `pa->capture` is null because the way it get's update is when a playback start.
I just submitted a [small PR](http://github.com/godotengine/godot/pull/37075/) that fix it, but I would like to confirm the expected behavior.

--- 

I was reading about Capture mode and it seems that its purpose is to allow an animation blend from the current state

#### Here is an example:

If I have an animation that change colors from red then green the blue, and every time I click on the screen I want to replay it.

1. Without capture mode: Its going abruptly change from current color (let's say greem) to red.

![Screen Recording 2020-03-15 at 15 26 29 (2)](https://user-images.githubusercontent.com/4673489/76708769-36685e00-66d8-11ea-8df1-db284a8c95ba.gif)

2. With capture mode: Its going to smoothly transition from current color (let's say green) to red.

![Screen Recording 2020-03-15 at 15 25 26 (2)](https://user-images.githubusercontent.com/4673489/76708767-35373100-66d8-11ea-9459-0e84c59c8b08.gif)

--- 

Based on this, for Capture Mode, the editor behavior when moving timeline cursor is not very relevant, because as long as you move it your are changing the starting state of the animation.
On Capture Mode, processing animation seems to be _Capturing_ the starting state (from where to begin the blending) right before playback is started (clicking play button for instance). If playback hasn't been started before, capture is null:

In `AnimationPlayer::_animation_process_animation` at `scene/animation/animation_player.cpp`
```
    if (p_started) {
        pa->capture = pa->object->get_indexed(pa->subpath);
    }
```

[So I added a check that allow to update the capture value if nothing has been captured before](https://github.com/godotengine/godot/pull/37075/commits/a6ed103550a3cd5dfe057a16bb97f4ce71a036e5#diff-5d3eb4d53ccdc2d87c6b0dfdea66c3e6L420):
```
    if (p_started || pa->capture.get_type() == Variant::NIL) {
        pa->capture = pa->object->get_indexed(pa->subpath);
    }
```

As this check is done in _animation_process_animation it has a consistent behavior when a user open the scene, because it's going to load the captured state with the frame that was selected as current when it was saved.

When objects are instantiated in the actual game, seems that the only way to run process animation is when animation is played, so this error doesn't appear.

I would really appreciate feedback on this so I can modify the PR if some of my assumptions are wrong :)

Thanks!

**Comment by akien-mga** [comment]
Fixed by #39329.
